### PR TITLE
f32: inline methods with special variant for msvc

### PR DESCRIPTION
**Pull request by pyfisch**

### CODE REVIEW

File: src/libstd/num/f32.rs [hunk 1]
```diff
@@ -664,13 +664,11 @@ impl f32 {
...
     pub fn log10(self) -> f32 {
-        return log10f(self);
-
         // see notes above in `floor`
         #[cfg(target_env = "msvc")]
-        fn log10f(f: f32) -> f32 { (f as f64).log10() as f32 }
+        return (f as f64).log10() as f32;
         #[cfg(not(target_env = "msvc"))]
-        fn log10f(f: f32) -> f32 { unsafe { intrinsics::log10f32(f) } }
+        return unsafe { intrinsics::log10f32(f) };
```
Comment (durka) [inline src/libstd/num/f32.rs]: `f` got left behind here

File: src/libstd/num/f32.rs [hunk 2]
```diff
@@ -557,13 +561,11 @@ impl f32 {
...
     pub fn exp(self) -> f32 {
-        return expf(self);
-
         // see notes above in `floor`
         #[cfg(target_env = "msvc")]
-        fn expf(f: f32) -> f32 { (f as f64).exp() as f32 }
+        return (self as f64).exp() as f32;
         #[cfg(not(target_env = "msvc"))]
-        fn expf(f: f32) -> f32 { unsafe { intrinsics::expf32(f) } }
+        return unsafe { intrinsics::expf32(self) };
```
Comment (durka) [inline src/libstd/num/f32.rs]: `f` got left behind here

File: src/libstd/num/f32.rs [hunk 3]
```diff
@@ -288,9 +296,9 @@ impl f32 {
         // redirect to this comment, so `floorf` is just one case of a missing
         // function on MSVC, but there are many others elsewhere.
         #[cfg(target_env = "msvc")]
-        fn floorf(f: f32) -> f32 { (f as f64).floor() as f32 }
+        return (self as f64).floor() as f32;
         #[cfg(not(target_env = "msvc"))]
-        fn floorf(f: f32) -> f32 { unsafe { intrinsics::floorf32(f) } }
+        return unsafe { intrinsics::floorf32(f) };
```
Comment (durka) [inline src/libstd/num/f32.rs]: `f` got left behind here

File: src/libstd/num/f32.rs [hunk 4]
```diff
@@ -664,13 +664,11 @@ impl f32 {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline]
     pub fn log10(self) -> f32 {
-        return log10f(self);
-
         // see notes above in `floor`
         #[cfg(target_env = "msvc")]
-        fn log10f(f: f32) -> f32 { (f as f64).log10() as f32 }
+        return (f as f64).log10() as f32;
```
Comment (durka) [inline src/libstd/num/f32.rs]: `f` got left behind here

**Comment by pyfisch** [comment]
Fixed float methods. Build passes.

**Comment by durka** [comment]
\o/

**Pull request merged**